x-pack/plugins/observability_ai_assistant/public/components/action_menu_item/action_menu_item.tsx

```diff
@@ -44,15 +47,15 @@ export function ObservabilityAIAssistantActionMenuItem() {
     };
   }, []);
 
+  const { from, to } = plugins.start.data.query.timefilter.timefilter.getTime();
   useEffect(() => {
-    const unregister = service.setScreenContext({
-      screenDescription: 'The user is looking at ' + window.location.href,
-    });
+    const start = datemath.parse(from)?.format();
+    const end = datemath.parse(to)?.format();
 
-    return () => {
-      unregister();
-    };
-  }, [service]);
+    return service.setScreenContext({
+      screenDescription: `The user is looking at ${window.location.href}. The current time range is ${start} - ${end}.`,
+    });
```

Comment:
Do we need to check for start/end to be non-nullish?

Reply:
In practice I don't see it happening but since the types allow it, probably a good idea. I'll add fallback values.

Reply:
Added in https://github.com/elastic/kibana/pull/176925/commits/4f9903d791a042d73846b4df47b24ac3ff2b222e

Reply:
I changed my mind and now require the time range to be set before specifying the range: https://github.com/elastic/kibana/pull/176925/commits/960693861a2e833dbb6d4c311fa836fba2204021

Reply:
Yeah I think it happens on the settings pages? Or does it e.g. return the default time range for Kibana? Either way this looks fine to me

Reply:
@sorenlouv ah but we still need to add the URL even if the time range is not defined

Reply:
True

Reply:
> @sorenlouv ah but we still need to add the URL even if the time range is not defined

I reverted back to https://github.com/elastic/kibana/pull/176925/commits/cbee239f982c5d902671f6b7a9a63f62eebabba1

Reply:
> Yeah I think it happens on the settings pages? Or does it e.g. return the default time range for Kibana? Either way this looks fine to me

On the settings page it uses the last specified time range (15m by default).